src/model.jl

```diff
@@ -116,43 +119,112 @@ function create_model(
     ## Objective function
     @objective(model, Min, assets_investment_cost + flows_investment_cost + flows_variable_cost)
 
+    function add_to_flow_sum_indexes!(flow_sum_indexes, a, rp, B, f)
+        for B_flow ∈ graph[f...].partitions[rp]
+            if B_flow[end] < B[1]
+                continue
+            end
+            d = duration(B, B_flow, rp)
+            if d != 0
+                push!(flow_sum_indexes, (d, f, B_flow))
+            end
+            if B_flow[1] > B[end]
+                break
+            end
+        end
+    end
+
+    flow_sum_indexes_incoming_lowest =
+        Dict{TupleAssetRPTimeBlock,Vector{TupleDurationFlowTimeBlock}}()
+    flow_sum_indexes_outgoing_lowest =
+        Dict{TupleAssetRPTimeBlock,Vector{TupleDurationFlowTimeBlock}}()
+    flow_sum_indexes_incoming_highest =
+        Dict{TupleAssetRPTimeBlock,Vector{TupleDurationFlowTimeBlock}}()
+    flow_sum_indexes_outgoing_highest =
+        Dict{TupleAssetRPTimeBlock,Vector{TupleDurationFlowTimeBlock}}()
+    for a ∈ A, rp ∈ RP
+        for B ∈ Pl[(a, rp)]
+            flow_sum_indexes_incoming_lowest[a, rp, B] = TupleDurationFlowTimeBlock[]
+            flow_sum_indexes_outgoing_lowest[a, rp, B] = TupleDurationFlowTimeBlock[]
+            for u ∈ inneighbor_labels(graph, a)
+                add_to_flow_sum_indexes!(
+                    flow_sum_indexes_incoming_lowest[a, rp, B],
+                    a,
+                    rp,
+                    B,
+                    (u, a),
+                )
+            end
+
+            for v ∈ outneighbor_labels(graph, a)
+                add_to_flow_sum_indexes!(
+                    flow_sum_indexes_outgoing_lowest[a, rp, B],
+                    a,
+                    rp,
+                    B,
+                    (a, v),
+                )
+            end
+        end
+        for B ∈ Ph[(a, rp)]
+            flow_sum_indexes_incoming_highest[a, rp, B] = TupleDurationFlowTimeBlock[]
+            flow_sum_indexes_outgoing_highest[a, rp, B] = TupleDurationFlowTimeBlock[]
+            for u ∈ inneighbor_labels(graph, a)
+                add_to_flow_sum_indexes!(
+                    flow_sum_indexes_incoming_highest[a, rp, B],
+                    a,
+                    rp,
+                    B,
+                    (u, a),
+                )
+            end
+
+            for v ∈ outneighbor_labels(graph, a)
+                add_to_flow_sum_indexes!(
+                    flow_sum_indexes_outgoing_highest[a, rp, B],
+                    a,
+                    rp,
+                    B,
+                    (a, v),
+                )
+            end
+        end
+    end
```

Comment:
Please don't take my word for it! I would be great if you could run the profile code to check for the speed up - so it isn't unilaterally informed by me. I can help with how to run it, if you want to do it, or you leave to Diego or Ni.
This is, allegedly, faster, because the `sum(... if ...)` structure inside the JuMP `@expression` is slow. It is faster to create the sets outside the loop, and then just use them. Furthermore, the loops inside the `@expression` were a double loop with non-contiguous values. Now, the loop inside the `@expression` is simpler.
The nested loops above are slow, but apparently not as slow as the older code.